core/promise/producer.go

```diff
@@ -0,0 +1,55 @@
+/*
+ * Copyright (C) 2017 The "MysteriumNetwork/node" Authors.
+ *
+ * This program is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU General Public License as published by
+ * the Free Software Foundation, either version 3 of the License, or
+ * (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with this program.  If not, see <http://www.gnu.org/licenses/>.
+ */
+
+package promise
+
+import (
+	"github.com/mysteriumnetwork/node/communication"
+)
+
+// Producer creates requests/responses of the "promise-create" events
+type Producer struct {
+	SignedPromise *SignedPromise
+}
+
+// Request structure represents message from service provider to receive new promise from consumer
+type Request struct {
+	SignedPromise *SignedPromise
```

Comment:
Why to have SignedPromise inside Request? Its just 1 member, why not simply SignedPromise?

Reply:
I think keeping `request/response` here, how it used in the communication dialog makes it more clear for understanding.

Reply:
Keep in mind, that will send SignedPromise structure to blockhain for clearing reasons